src/search.c

```diff
@@ -263,12 +382,54 @@ static bool file_search(char *filename, char *path, char *search_term)
 	return false;
 }
 
-/* report_find()
+double guess_score(char *bundle, char *file, char *search_term)
+{
+	double multiplier = 1.0;
+	double score = 1.0;
+
+	char bnb[PATH_MAX];
+	char dnb[PATH_MAX];
+
+	char *bn;
+	char *dn;
+
+	strcpy(bnb, file);
+	strcpy(dnb, file);
+
+	bn = basename(bnb);
+	dn = dirname(dnb);
+
+	if (strstr(bundle, "-dev"))
+		multiplier /= 10;
+
+	if (strcmp(dn, "/usr/bin") == 0)
+		score += 5;
+
+	if (strcmp(dn, "/usr/include") == 0)
+		multiplier /= 2;
+
+	if (strstr(bundle, search_term))
+		score += 0.5;
+
+	/* slightly favor shorter filenames */
+	score += (2.0 / strlen(file));
+
+	if (strcmp(bn, search_term) == 0)
+		multiplier *= 5;
+
+	return multiplier * score;
+}
+
+/* report_finds()
  * Report out, respecting verbosity
  */
-static void report_find(char *bundle, char *file)
+static void report_find(char *bundle, char *file, char *search_term, struct manifest *man)
 {
-	printf("'%s'  :  '%s'\n", bundle, file);
+	double score;
+
+	score = guess_score(bundle, file, search_term);
+	//	printf("'%s'  :  '%s'   (%5.1f)\n", bundle, file, score);
```

Comment:
This was the old way of doing swupd search. I think a flag for swupd search would be helpful to toggle `--all-results` and just print the flat list. Makes it more machine-readable.